### PR TITLE
Implement simple functions

### PR DESCRIPTION
**Pull request by nadavge**
I've been waiting for you, and decided it would be better if I implemented some functions myself.

### CODE REVIEW

File: CachingFileSystem.cpp [hunk 1]
```diff
@@ -22,12 +24,18 @@ using namespace std;
 // Used to hold the root path for OS file reads
 // TODO fill this with the full path using *realpath(3)*
 static char gRootpath[PATH_MAX] = {0};
```
Comment (BestTeamEver) [inline CachingFileSystem.cpp]: Maybe gRootPath?

File: CachingFileSystem.cpp [hunk 2]
```diff
@@ -22,12 +24,18 @@ using namespace std;
...
+// TODO when filling the rootpath, enter its length in here
+static size_t gRootpathLength = 0;
```
Comment (BestTeamEver) [inline CachingFileSystem.cpp]: Possibly gRootPathLength?

File: CachingFileSystem.cpp [hunk 3]
```diff
@@ -36,72 +44,119 @@ static size_t gBlockSize = 0;
 
 // ================================= IMPLEMENTATION ================================ //
 
-int caching_getattr(const char *path, struct stat *statbuf){
-	// TODO Straight out of the box (hereinafter: SOOTB)
-	return 0;
+static void getFullPath(char fullPath[PATH_MAX], const char *path)
+{
+	if (strlen(path) + gRootpathLength >= PATH_MAX)
```
Comment (BestTeamEver) [inline CachingFileSystem.cpp]: Add realpath?

File: CachingFileSystem.cpp [hunk 4]
```diff
@@ -36,72 +44,119 @@ static size_t gBlockSize = 0;
 
 // ================================= IMPLEMENTATION ================================ //
 
-int caching_getattr(const char *path, struct stat *statbuf){
-	// TODO Straight out of the box (hereinafter: SOOTB)
-	return 0;
+static void getFullPath(char fullPath[PATH_MAX], const char *path)
```
Comment (BestTeamEver) [inline CachingFileSystem.cpp]: should return something to indicate success/fail?

File: CachingFileSystem.cpp (hunk 1 above)
Comment (nadavge) [inline CachingFileSystem.cpp]: Agreed

File: CachingFileSystem.cpp (hunk 2 above)
Comment (nadavge) [inline CachingFileSystem.cpp]: Yep

File: CachingFileSystem.cpp (hunk 3 above)
Comment (nadavge) [inline CachingFileSystem.cpp]: This should be done in the initialization, as it doesn't change. Thats what RootPath is for

File: CachingFileSystem.cpp (hunk 4 above)
Comment (nadavge) [inline CachingFileSystem.cpp]: Yes, in the forum they state to return -ENAMETOOLONG

http://moodle.cs.huji.ac.il/cs14/mod/forum/discuss.php?d=5398

**Pull request merged**